src/Controller/Api/GenreController.php

```diff
@@ -0,0 +1,50 @@
+<?php
+namespace App\Controller\Api;
+
+use App\Entity\Genre;
+use Doctrine\ORM\EntityManagerInterface;
+use FOS\RestBundle\Controller\AbstractFOSRestController;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use FOS\RestBundle\Request\ParamFetcher;
+use FOS\RestBundle\View\View;
+use Knp\Component\Pager\PaginatorInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+
+/**
+ * Class GenreController
+ * @package App\Controller\Api
+ *
+ */
+class GenreController extends AbstractFOSRestController
+{
+    /**
+     * @var EntityManagerInterface|null
+     */
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
```

Comment:
And here we are breaking the boundaries again. EntityManager should only exist in the Persistence layer and never in the controller.

Reply:
Unfortunately I should have implemented like I did in BookController where I moved the entity manager in book manager service